src/app/teacher-round.service.ts

```diff
@@ -42,29 +50,24 @@ export class TeacherRoundService {
    * Create a new round within the current session, mark it as the currently
    * active round, set its initial state, and hook up the TimerService so that
    * when the timer ticks, the round updates.
-   *
-   * TODO: As of this iteration, this function just re-uses a single demo
-   * round and re-populates it with data, rather than creating a new round.
-   * We also don't bother with marking it as the current round.
    */
   // In the future, we might get sessionId from a TeacherSessionService, rather
   // than passing it in as a parameter.
-  startNewRound(sessionId: string, roundData: FirebaseRound): void {
-    // Eventually, we'll create a new round, but for the moment, we'll just use
-    // this one.
-    this.roundPath$.next(demoRoundPath);
-    this.firebaseService.setRoundData(demoRoundPath, roundData);
-  }
+  startNewRound(roundData: FirebaseRound): void {
+    this.teacherSessionService.sessionId$.pipe(take(1)).subscribe(sessionId => {
+      this.firebaseService.createRoundInSession(sessionId, roundData).then(roundPath => {
+        this.firebaseService.setCurrentRound(roundPath).then(() => {
+          this.teacherSessionService.currentRoundPath$.next(roundPath); });
+      });
```

Comment:
The indentation and bracketing here seems a bit off

Reply:
```suggestion
    this.teacherSessionService.sessionId$.pipe(take(1)).subscribe(sessionId => {
      this.firebaseService.createRoundInSession(sessionId, roundData).then(roundPath => {
        this.firebaseService.setCurrentRound(roundPath).then(() => {
          this.teacherSessionService.currentRoundPath$.next(roundPath);
        });
      });
```

How about this?